db/version_set.cc

```diff
@@ -5518,6 +5524,14 @@ Status VersionSet::ProcessManifestWrites(
                             dir_contains_current_file);
       if (!io_s.ok()) {
         s = io_s;
+        // TODO: when CURRENT file rename fails, should the in memory state
+        // continue to be updated to use the new descriptor, should we do:
+        // manifest_io_status = io_s;
```

Comment:
I don't know. The CURRENT rename might have succeeded and returned an error, in which case a restart will look for the new MANIFEST. But if we set `manifest_io_status` to non-OK it looks like it cleans up the new MANIFEST. I think keeping both like you're doing below is safest although if there's a cleaner way to do it correctly that'd be a welcome improvement.

Reply:
Thank you for the detailed context. I agree that keeping both is safest. I removed this TODO since there isn't too much of a follow up needed here.